File: 283. Move Zeroes/283. Move Zeroes.go
func moveZeroes(nums []int)  {
    var cnt = 0
    var j = 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != 0 {
            nums[j] = nums[i]
            j++
        } else {
            cnt++
        }
    }
    for ; cnt > 0; cnt-- {
        nums[len(nums) - cnt] = 0
    }
    return
}

File: 283. Move Zeroes/283. Move Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    var j = 0
    var cnt = 0
    for (var i = 0; i < nums.length; i++) {
        if (nums[i] != 0) {
            nums[j++] = nums[i]
        } else {
            cnt++
        }
    }
    while (cnt > 0) {
        nums[nums.length - cnt--] = 0
    }
    return
};
